enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/KeyValidator.java

```diff
@@ -55,17 +59,21 @@ public class KeyValidator {
      *
      * @param validationContext the token validation context
      * @return true is the scopes are valid
-     * @throws EnforcerException if key validation information not set
+     * @throws EnforcerException this will indicate the message body for the error response
      */
-    public static boolean validateScopes(TokenValidationContext validationContext) throws EnforcerException {
+    public static boolean validateScopes(TokenValidationContext validationContext) throws APISecurityException {
 
         if (validationContext.isCacheHit()) {
             return true;
         }
         APIKeyValidationInfoDTO apiKeyValidationInfoDTO = validationContext.getValidationInfoDTO();
 
         if (apiKeyValidationInfoDTO == null) {
-            throw new EnforcerException("Key Validation information not set");
+            log.error("Error while accessing backend services for token scope validation",
```

Comment:
```suggestion
            log.error("Error while validating scopes. Key validation information has not been set.",
```

Reply:
You have suggested this exact code here https://github.com/wso2-support/product-microgateway/pull/217#discussion_r930817765, do you want me to change it?

Reply:
😄 yes, shall we?

Reply:
Then, it would be precisely the same as the throwing APISecurityException, so we can just remove this log. That's what you meant right? @suksw

Reply:
However, that error log looks acceptable to me. Do we really need to change it?

Reply:
No just to update the log to mention that it is a scope validation error.

Reply:
*without (sorry about that)